lib/filter.rb

```diff
@@ -89,7 +89,7 @@ def add_element_filter_from_definition(filter_definition)
     new_element_filters = Filter.parse_filter_definitions(filter_definition)
 
     new_element_filters.each do |path, element_filter|
-      @element_filters[path] ||= ElementFilter.new(path)
+      @element_filters[path] ||= ElementFilter.new(path, element_filter.operator)
       @element_filters[path].add_matchers(element_filter.matchers)
```

Comment:
This will fail, when there are element filters for the same path with different operators.